satpy/readers/seviri_l1b_native.py

```diff
@@ -483,51 +483,45 @@ def get_dataset(self, dataset_id, dataset_info):
         # even though all the channels may not be present in the file,
         # the header does have calibration coefficients for all the channels
         # hence, this channel index needs to refer to full channel list
-        i = list(CHANNEL_NAMES.values()).index(channel)
-
-        if calibration == 'counts':
-            return data
-
-        if calibration in ['radiance', 'reflectance', 'brightness_temperature']:
-            # determine the required calibration coefficients to use
-            # for the Level 1.5 Header
-            if (self.calib_mode.upper() != 'GSICS' and self.calib_mode.upper() != 'NOMINAL'):
-                raise NotImplementedError(
-                    'Unknown Calibration mode : Please check')
-
-            # NB GSICS doesn't have calibration coeffs for VIS channels
-            if (self.calib_mode.upper() != 'GSICS' or channel in VIS_CHANNELS):
-                coeffs = data15hdr[
-                    'RadiometricProcessing']['Level15ImageCalibration']
-                gain = coeffs['CalSlope'][i]
-                offset = coeffs['CalOffset'][i]
-            else:
-                coeffs = data15hdr[
-                    'RadiometricProcessing']['MPEFCalFeedback']
-                gain = coeffs['GSICSCalCoeff'][i]
-                offset = coeffs['GSICSOffsetCount'][i]
-                offset = offset * gain
-            res = self._convert_to_radiance(data, gain, offset)
-
-        if calibration == 'reflectance':
-            solar_irradiance = CALIB[self.platform_id][channel]["F"]
-            res = self._vis_calibrate(res, solar_irradiance)
-
-        elif calibration == 'brightness_temperature':
-            cal_type = data15hdr['ImageDescription'][
-                'Level15ImageProduction']['PlannedChanProcessing'][i]
-            res = self._ir_calibrate(res, channel, cal_type)
-
-        logger.debug("Calibration time " + str(datetime.now() - tic))
-        return res
+        band_idx = list(CHANNEL_NAMES.values()).index(channel_name)
+
+        coefs_nominal = self.header['15_DATA_HEADER'][
+            'RadiometricProcessing']['Level15ImageCalibration']
+        coefs_gsics = self.header['15_DATA_HEADER'][
+            'RadiometricProcessing']['MPEFCalFeedback']
+        radiance_types = self.header['15_DATA_HEADER']['ImageDescription'][
+                'Level15ImageProduction']['PlannedChanProcessing']
+        return {
+            'coefs': {
+                'NOMINAL': {
+                    'gain': coefs_nominal['CalSlope'][band_idx],
+                    'offset': coefs_nominal['CalOffset'][band_idx]
+                },
+                'GSICS': {
+                    'gain': coefs_gsics['GSICSCalCoeff'][band_idx],
+                    'offset': coefs_gsics['GSICSOffsetCount'][band_idx]
+                },
+                'EXTERNAL': self.ext_calib_coefs.get(channel_name, {})
+            },
+            'radiance_type': radiance_types[band_idx]
+        }
```

Comment:
These lines are very similar to the ones in hrit...

Reply:
I'll make a helper method for this